src/components/ChatEntry.js

```diff
@@ -1,22 +1,28 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp'
 
-const ChatEntry = (props) => {
+const ChatEntry = ({ id, sender, liked, body,timeStamp, handleLike }) => {
+
+  const heart = liked ? '❤️' : '🤍';
+  
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={`chat-entry ${id % 2 !== 0 ? 'local' : 'remote'}`}>
+      <h2 className="entry-name">{sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{body}</p>
+        <TimeStamp className="entry-time" time={timeStamp}/>
+        <button className="like" onClick={() => handleLike(id)}>{heart}</button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+    sender: PropTypes.string.isRequired,
+    body: PropTypes.string.isRequired,
+    timeStamp: PropTypes.string.isRequired,
```

Comment:
Seems like you forgot a couple of props! This is okay, but make sure to double check! Good job on the ones that you did implement!